src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -193,64 +210,182 @@ We support
+            {
+                IEnumerable<IConfigurationSection> filterSections = conditions.GetSection(AzureAppConfigurationFeatureFlagConfigurationFields.ClientFilters).GetChildren();
+
+                if (filterSections.Any())
+                {
+                    foreach (IConfigurationSection section in filterSections)
+                    {
+                        //
+                        // Arrays in json such as "myKey": [ "some", "values" ]
+                        // Are accessed through the configuration system by using the array index as the property name, e.g. "myKey": { "0": "some", "1": "values" }
+                        if (int.TryParse(section.Key, out int _) && !string.IsNullOrEmpty(section[ConfigurationFields.NameKeyword]))
+                        {
+                            enabledFor.Add(new FeatureFilterConfiguration()
+                            {
+                                Name = section[ConfigurationFields.NameKeyword],
+                                Parameters = new ConfigurationWrapper(section.GetSection(ConfigurationFields.FeatureFilterConfigurationParameters))
+                            });
+                        }
+                    }
+                }
+                else
+                {
+                    enabledFor.Add(new FeatureFilterConfiguration
+                    {
+                        Name = "AlwaysOn"
+                    });
+                }
+            }
+
+            return new FeatureDefinition()
+            {
+                Name = featureName,
+                EnabledFor = enabledFor,
+                RequirementType = requirementType
+            };
+        }
+
+        private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
+        {
+            IConfigurationSection featureManagementConfigurationSection = _configuration.GetSection(ConfigurationFields.FeatureManagementSectionName);
+
+            if (!featureManagementConfigurationSection.Exists())
+            {
+                if (!RootConfigurationFallbackEnabled)
+                {
+                    Logger?.LogDebug($"No configuration section named '{ConfigurationFields.FeatureManagementSectionName}' was found.");
+
+                    return Enumerable.Empty<IConfigurationSection>();
+                }
+
+                featureManagementConfigurationSection = _configuration as IConfigurationSection;
+            }
+
+            IConfigurationSection featureFlagsConfigurationSection = featureManagementConfigurationSection.GetSection(AzureAppConfigurationFeatureFlagConfigurationFields.FeatureFlagsSectionName);
+
             //
-            // There is no "FeatureManagement" section in the configuration
-            if (RootConfigurationFallbackEnabled)
+            // "FeatureFlag" section should be an array if Azure App Configuration feature flag schema is enabled
+            _azureAppConfigurationFeatureFlagSchemaEnabled = featureFlagsConfigurationSection.Exists() &&
```

Comment:
I left another comment about this. This is not designed well to be assigned to multiple times. This class is designed as thread safe. Meaning this value can change while another thread is somewhere having previously evaluated it.

Either a bigger redesign is needed, or we need to assign to this only once.

Reply:
The latest update does not solve the problem.

Once a public method is invoked and takes an action based off the schema, it should stick with that schema the entire time. Otherwise unexpected things will happen.

Originally I was thinking to just limit the schema to not be able to change during the runtime of the application. Is this acceptable?

If not, then we can have to separate internal feature definition providers that we invoke in the ConfigurationFeatureDefinitionProvider based off of the detected schema.
* DefaultConfigurationFeatureDefinitionProvider
* AacConfigurationFeatureDefinitionProvider

Reply:
```
if (_aacProvider.HasFlags)
{
    _aacProvider.Xyz();
}
else
{
    _defaultProvider.Xyz();
}
```

---

_configuration.OnChange(() =>
{
    _aacProvider.MarkStale();

    _defaultProvider.MarkStale();
});

Reply:
> Originally I was thinking to just limit the schema to not be able to change during the runtime of the application. Is this acceptable?
It depends on whether we consider the scenario that configuration is dynamically changed to a different schema during the runtime, is a valid user case.

Richard mentioned this potential user case:
https://github.com/microsoft/FeatureManagement-Dotnet/pull/319#discussion_r1408827330

I am thinking whether we should allow both schema to be recognized at the same time.